pypowsybl/perunitnetwork.py

```diff
@@ -0,0 +1,392 @@
+        df['target_p'] *= self.sn
+        df['target_q'] *= self.sn
+        df = df.drop(['nominal_v', 'voltage_level_id'], axis=1)
+        return self.update_elements(_pypowsybl.ElementType.GENERATOR, df)
+
+    def update_loads(self, df: pd.DataFrame):
+        """ Update loads with a ``Pandas`` data frame.
+
+        Args:
+            df (DataFrame): the ``Pandas`` data frame
+        """
+        df['p0'] *= self.sn
+        df['q0'] *= self.sn
+        return self.update_elements(_pypowsybl.ElementType.LOAD, df)
+
+    def update_batteries(self, df: pd.DataFrame):
+        """ Update batteries with a ``Pandas`` data frame.
+
+        Available columns names:
+        - p0
+        - q0
+        Args:
+            df (DataFrame): the ``Pandas`` data frame
+        """
+        df['p0'] *= self.sn
+        df['q0'] *= self.sn
+        return self.update_elements(_pypowsybl.ElementType.BATTERY, df)
+    
+    def update_dangling_lines(self, df: pd.DataFrame):
+        """ Update dangling lines with a ``Pandas`` data frame.
+
+        Args:
+            df (DataFrame): the ``Pandas`` data frame
+        """
+        df['p0'] *= self.sn
+        df['q0'] *= self.sn
+        return self.update_elements(_pypowsybl.ElementType.DANGLING_LINE, df)
+    
+    def update_vsc_converter_stations(self, df: pd.DataFrame):
+        """ Update VSC converter stations with a ``Pandas`` data frame.
+
+        Args:
+            df (DataFrame): the ``Pandas`` data frame
+        """
+        df['reactive_power_setpoint'] *= self.sn
+        df = pd.merge(df, self.get_vsc_converter_stations()['voltage_level_id'],
+                      left_index=True, right_index=True)
+        df = pd.merge(df, self.get_voltage_levels()['nominal_v'], left_on='voltage_level_id', right_index=True)
+        df['voltage_setpoint'] *= df['nominal_v']
+        df = df.drop(['nominal_v', 'voltage_level_id'], axis=1)
+        return self.update_elements(_pypowsybl.ElementType.VSC_CONVERTER_STATION, df)
+    
+    def update_static_var_compensators(self, df: pd.DataFrame):
+        """ Update static var compensators with a ``Pandas`` data frame.
+
+        Args:
+            df (DataFrame): the ``Pandas`` data frame
+        """
+        df['reactive_power_setpoint'] *= self.sn
+        df = pd.merge(df, self.get_static_var_compensators()['voltage_level_id'],
+                      left_index=True, right_index=True)
+        df = pd.merge(df, self.get_voltage_levels()['nominal_v'], left_on='voltage_level_id', right_index=True)
+        df['voltage_setpoint'] *= df['nominal_v']
+        df = df.drop(['nominal_v', 'voltage_level_id'], axis=1)
+        return self.update_elements(_pypowsybl.ElementType.STATIC_VAR_COMPENSATOR, df)
+    
+    def update_hvdc_lines(self, df: pd.DataFrame):
+        """ Update HVDC lines with a ``Pandas`` data frame.
+
+        Args:
+            df (DataFrame): the ``Pandas`` data frame
+        """
+        df['active_power_setpoint'] *= self.sn
```

Comment:
in all update methods:
this kind of call will change the user dataframe "in place", which we should probably not do, because he will not expect it.
For example if you call twice this update method with the same dataframe, the second time it will do something weird !